pyninjotiff/ninjotiff.py

```diff
@@ -308,14 +313,33 @@ def _finalize(img, dtype=np.uint8, value_range_measurement_unit=None, data_is_sc
     **Notes**:
         physic_val = image*scale + offset
         Example values for value_range_measurement_unit are (0, 125) or (40.0, -87.5)
+
+    ***Warning***
+        Only the 'L' and 'RGB' cases are compatible with xarray.XRImage.
+        They still have to  be tested thoroughly.
     """
+    if type(img) == trollimage.xrimage.XRImage:
+        is_masked = False
+        log.debug("Latest version using trollimage.xrimage.XRImage")
+
     if img.mode == 'L':
         # PFE: mpop.satout.cfscene
-        data = img.channels[0]
-        fill_value = np.iinfo(dtype).min
-        log.debug("Transparent pixel are forced to be %d" % fill_value)
+        if is_masked :
+            data = img.channels[0]
+        else :
+            # TODO: check what is the corret fill value for NinJo!
+            if fill_value != None:
+                log.debug("Forcing fill value to "+str(fill_value))
+            data = img.finalize(dtype=dtype, fill_value=fill_value)
+            # Go back to the masked_array for compatibility
+            # with the following part of the code.
+            data = data[0].to_masked_array()
+
+        fill_value = fill_value or np.iinfo(dtype).min
```

Comment:
This should probably do `fill_value if fill_value is not None else np.iinfo(dtype).min` since a `fill_value` of 0 is considered "False".